astropy/coordinates/baseframe.py

```diff
@@ -137,168 +137,14 @@ def _normalize_representation_type(kwargs):
-            RepresentationMapping('d_lat',
-                                  'pm_{lat}'.format(**sph_component_map),
-                                  u.mas/u.yr),
-            RepresentationMapping('d_distance', 'radial_velocity',
-                                  u.km/u.s)
-        ])
-
-        repr_info.setdefault(r.SphericalDifferential, [
-            RepresentationMapping('d_lon',
-                                  'pm_{lon}'.format(**sph_component_map),
-                                  u.mas/u.yr),
-            RepresentationMapping('d_lat',
-                                  'pm_{lat}'.format(**sph_component_map),
-                                  u.mas/u.yr),
-            RepresentationMapping('d_distance', 'radial_velocity',
-                                  u.km/u.s)
-        ])
-
-        repr_info.setdefault(r.CartesianDifferential, [
-            RepresentationMapping('d_x', 'v_x', u.km/u.s),
-            RepresentationMapping('d_y', 'v_y', u.km/u.s),
-            RepresentationMapping('d_z', 'v_z', u.km/u.s)])
-
-        # Unit* classes should follow the same naming conventions
-        # TODO: this adds some unnecessary mappings for the Unit classes, so
-        # this could be cleaned up, but in practice doesn't seem to have any
-        # negative side effects
-        repr_info.setdefault(r.UnitSphericalRepresentation,
-                             repr_info[r.SphericalRepresentation])
-
-        repr_info.setdefault(r.UnitSphericalCosLatDifferential,
-                             repr_info[r.SphericalCosLatDifferential])
-
-        repr_info.setdefault(r.UnitSphericalDifferential,
-                             repr_info[r.SphericalDifferential])
-
-        # Make read-only properties for the frame class attributes that should
-        # be read-only to make them immutable after creation.
-        # We copy attributes instead of linking to make sure there's no
-        # accidental cross-talk between classes
-        mcls.readonly_prop_factory(members, 'default_representation',
-                                   default_repr)
-        mcls.readonly_prop_factory(members, 'default_differential',
-                                   default_diff)
-        mcls.readonly_prop_factory(members,
-                                   'frame_specific_representation_info',
-                                   copy.deepcopy(repr_info))
-
-        # now set the frame name as lower-case class name, if it isn't explicit
-        if 'name' not in members:
-            members['name'] = name.lower()
-
-        # A cache that *must be unique to each frame class* - it is
-        # insufficient to share them with superclasses, hence the need to put
-        # them in the meta
-        members['_frame_class_cache'] = {}
-
-        return super().__new__(mcls, name, bases, members)
-
-    @staticmethod
-    def readonly_prop_factory(members, attr, value):
-        private_attr = '_' + attr
+def _readonly_prop_factory(cls, attr_name, value):
```

Comment:
My tendency would be to change this to a private `@classmethod` instead of a stand-alone function (and maybe call it `_create_readonly_property`?)

Reply:
Also no strong opinion here, so I made the change!